src/texture-processor/detail/accessor.hh

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include <typed-geometry/tg-lean.hh>
+
+#include <texture-processor/extents.hh>
+
+// TODO: variable names could be even better
+// TODO: add other functions like contains
+namespace tp::detail
+{
+template <int D, class this_t>
+struct accessor;
```

Comment:
Maybe add a comment explaining that `image_view` inherits these for conditional inheritance-style .at(x,y) / operator()(x,y)